library/src/main/kotlin/build/buf/connect/http/HTTPClientInterface.kt

```diff
@@ -56,8 +56,12 @@ class Stream(
         if (isClosed()) {
             return Result.failure(IllegalStateException("cannot send. underlying stream is closed"))
         }
-        onSend(buffer)
-        return Result.success(Unit)
+        return try {
```

Comment:
This is the contract with the underlying network -- we expect `onSend` to be unsafe and will handle that failure appropriately.